corpora-api.yml

```diff
@@ -0,0 +1,530 @@
+        returned with the project’s UUID.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                needs_attestation:
+                  type: boolean
+                  description: >-
+                    Determines whether this project requires attestation in which case this project cannot be submitted
+                    without a Terms and Conditions file also uploaded.
+                project_uuid:
+                  type: string
+                  description: >-
+                    If the submission is created in order to update an existing publicly published project, then a
+                    Project UUID must be provided in order to backfill the new submission entity with the existing
+                    project details for modification.
+              required:
+                - needs_attestation
+      responses:
+        '201':
+          description: A new submissions has been created.
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  project_uuid:
+                    $ref: '#/components/schemas/project_uuid'
+        '400':
+          $ref: '#/components/responses/400'
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+
+
+  /v1/submission/{project_uuid}:
+
+    get:
+      summary: Get a project's submission details by UUID
+      tags:
+        - submissions
+      description: >-
+        Returns all available metadata information about a project submission, including URIs of datasets that are
+        attached to the project.
+      parameters:
+        - $ref: '#/components/parameters/path_project_uuid'
+      responses:
+        '200':
+          $ref: '#/components/responses/200'
+        '400':
+          $ref: '#/components/responses/400'
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+
+    delete:
+      summary: Delete a submission
+      tags:
+        - submissions
+      description: >-
+        Deletes the submission associated with the given project UUID. This does not delete the project if the project
+        associated with the submission has been previously publicly published.
+      parameters:
+        - $ref: '#/components/parameters/path_project_uuid'
+      responses:
+        '202':
+          $ref: '#/components/responses/202'
+        '400':
+          $ref: '#/components/responses/400'
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+
+
+  /v1/submission/{project_uuid}/file:
```

Comment:
This endpoint accepts 3 body parameters: `filename`, `filetype`, `body`. Is the `body` intended to contain the encoded contents of the uploaded file? If so, I was wondering if we would consider an alternative approach to file upload, possible options proposed [here](https://stackoverflow.com/a/4083908).

In particular, I'm interested in the two-part upload option: 1) POST `/file_metadata` to create a file entry, and 2) POST `/file` via `multipart/form-data`, as a cleaner and possibly faster file upload pattern. Thoughts @Bento007 and @maniarathi ? Curious how the DCP Upload Service performed this operation as well.